mkosi/backend.py

```diff
@@ -323,7 +338,7 @@ def partition_path(self, ident: PartitionIdentifier, loopdev: Path) -> Optional[
 
     def sfdisk_spec(self) -> str:
         table = ["label: gpt",
-                 f"grain: {self.grain}",
+                 f"grain: {self.grain}" if sfdisk_grain_is_supported() else '\n',
```

Comment:
Having two calls to `sfdisk_grain_is_supported` is a bit unfortunate, but I guess it's still a better option than the alternative of checking whether the grain is `1024**2`, what do you think?

Reply:
Maybe caching `sfdisk_grain_is_supported`'s result would help here? Basically having a "static" (whatever that translates to python) tristate variable, which would initialize itself on the first run of the function and would return its result on all subsequent calls without the need to call `sfdisk`.

Reply:
This could be done most simply by adding another attribute `_sfdisk_has_grain: bool = False` to the dataclass and setting it in `__post_init__`.

Reply:
I went with `functools.lru_cache()` on `sfdisk_grain_is_supported()` to memoize the result.